packages/website/src/components/ESQueryFilter.tsx

```diff
@@ -0,0 +1,39 @@
+import type { Selector } from 'esquery';
+import React, { useState } from 'react';
+
+import styles from './ESQueryFilter.module.css';
+import Text from './inputs/Text';
+
+export interface ESQueryFilterProps {
+  readonly onChange: (value: Selector) => void;
+  readonly onError: (value?: Error) => void;
+}
+
+export function ESQueryFilter({
+  onChange,
+  onError,
+}: ESQueryFilterProps): JSX.Element {
+  const [value, setValue] = useState('');
+  const changeEvent = (value: string): void => {
+    setValue(value);
+    try {
+      const queryParsed = window.esquery.parse(value);
+      onChange(queryParsed);
+      onError(undefined);
+    } catch (e: unknown) {
+      onError(e instanceof Error ? e : new Error(String(e)));
```

Comment:
I bet a lot of users (myself included, heh) will be typing incorrect queries - at least when they're not done typing yet. It's a little off-putting to see _Internal error_ show up for that state too.

<img width="583" alt="Screenshot of an 'invalid selector' error with a syntax parsing message and full stack trace" src="https://user-images.githubusercontent.com/3335181/223866267-d240fac1-3c0a-44e6-bac4-fee3042085ea.png">

Taking a step back, this isn't really an _error_ state. It's a known state that we expect users to hit frequently. And I don't think we need to capture two states for the result of parsing: it's either a happy result (`onChange` & `Selector`) or parsing error (`onError` & right now, `Error`). How about merging them into one state?

* Combining `onChange` and `onError` into `onChange: (value: string | Selector)`
* Treating a `value: string` as a warning rather than an error

Mocking up how that'd look:

<img width="578" alt="Screenshot of an 'invalid selector' warning with the same error message, but no stack trace" src="https://user-images.githubusercontent.com/3335181/223867029-674eeff5-cdbc-4b29-ba43-36f2b32009a3.png">